File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    answer = []
    
    dict = {"zero" : 0, "one" : 1, "two" : 2, "three" : 3, "four" : 4, "five" : 5, "six" : 6, "seven" : 7, "eight" : 8, "nine" : 9}
    s = list(s) # 문자열 배열로 변환
    text = ""
    
    for x in range(len(s)):
        # 알파벳인지 확인
        if s[x].isalpha():
            text = text+s[x]
            if text in dict:
                answer.append(dict[text])
                text = ""
        else:
            # 알파벳 아니면 바로 추가
            answer.append(s[x])
        
        result = ''.join(str(s) for s in answer)
        
    return int(result)

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    const words = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"];
    
    words.forEach((word, i) => {
        s = s.replaceAll(word, i);
    });
    
    return parseInt(s);
}